### PR TITLE
log truncation integrated

### PR DESCRIPTION
**Pull request by mmucklo**
Builds, passes paxos tests.

TODO(mmucklo): needs more testing.

### CODE REVIEW

File: paxos/node.cc [hunk 1]
```diff
@@ -77,30 +78,86 @@ void PaxosNode::TruncationLoop(std::stop_token st) {
...
+      {
+        absl::ReaderMutexLock l(&lock_);
+        acceptor_stubs_size = acceptor_stubs_.size();
       }
+      uint64_t min_index = std::numeric_limits<uint64_t>::max();
+      bool skip = false;
+      for (size_t i = 0; i < acceptor_stubs_size; i++) {
+        if (!IsLeader()) {
+          break;
+        }  // small optimization, but as explained above still okay if we don't
```
Comment (vramesh7) [inline paxos/node.cc:96]: Looks like the comment is in the wrong place ?

File: paxos/node.cc [hunk 2]
```diff
@@ -77,30 +78,86 @@ void PaxosNode::TruncationLoop(std::stop_token st) {
...
+        request.set_index(min_index);
+        paxos_rpc::TruncateResponse response;
+        grpc::ClientContext context;
+        absl::ReaderMutexLock l(&lock_);
+        if (stub == nullptr) {
+          // We could still truncate as we've heard back from everyone.
+          // Just the down node will not be able to truncate, which is okay
+          // it will truncate in another round.
+          LOG(INFO) << "Skipping truncation of node: " << i
+                    << " because it offline.";
```
Comment (vramesh7) [inline paxos/node.cc:149]: grammar

**Review by vramesh7 — APPROVED** [review]
Couple of nits

File: paxos/node.cc (hunk 1 above)
Comment (mmucklo) [reply paxos/node.cc]: fixed.

File: paxos/node.cc (hunk 2 above)
Comment (mmucklo) [reply paxos/node.cc:157]: fixed.

File: paxos/node.cc (hunk 1 above)
Comment (mmucklo) [reply paxos/node.cc]: fixed.

**Pull request merged**